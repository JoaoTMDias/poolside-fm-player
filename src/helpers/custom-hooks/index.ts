export * from "./use-event";
export * from "./use-lifecycle-hooks";
export * from "./use-debounced-effect";
