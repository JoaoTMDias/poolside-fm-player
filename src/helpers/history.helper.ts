import { createBrowserHistory } from "history";

export const history = createBrowserHistory({
	basename: process.env.REACT_APP_BASENAME,
});

export default history;
