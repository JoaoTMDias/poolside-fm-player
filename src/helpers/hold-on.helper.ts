/**
 * Returns a promise that resolves after how many milliseconds you pass it.
 *
 * @author Jo√£o Dias
 * @param {number} [amount=0]
 * @returns
 */
export function holdOn(amount = 0): Promise<NodeJS.Timeout> {
	return new Promise((resolve) => setTimeout(resolve, amount));
}

export default holdOn;
