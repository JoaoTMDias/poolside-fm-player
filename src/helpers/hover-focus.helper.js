import { css } from "styled-components";

export const hoverFocus = styles =>
	css`
		&:hover:enabled,
		&:focus:enabled {
			${css(...styles)}
		}
	`;

export default hoverFocus;
