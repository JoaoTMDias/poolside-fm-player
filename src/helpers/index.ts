import hasPressedSpaceOrEnter from "./has-pressed-space-or-enter";

export * from "./logger.helper";
export * from "./key-codes";
export * from "./with-memo.helper";
export * from "./hold-on.helper";
export * from "./hover-focus.helper";
export * from "./history.helper";

export { hasPressedSpaceOrEnter };
