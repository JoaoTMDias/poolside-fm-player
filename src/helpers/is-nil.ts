export const isNil = (val: any) => val == null;
