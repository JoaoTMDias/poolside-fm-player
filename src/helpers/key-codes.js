export const KEY_CODES = {
	BACKSPACE: 8,
	TAB: 9,
	ENTER: 13,
	OK: 13,
	SHIFT: 16,
	ESC: 27,
	SPACE: 32,
	END: 35,
	HOME: 36,
	ARROW_LEFT: 37,
	ARROW_UP: 38,
	ARROW_RIGHT: 39,
	ARROW_DOWN: 40,
	ONE: 49,
	TWO: 50,
	THREE: 51,
	FOUR: 52,
	FIVE: 53,
	SIX: 54,
	SEVEN: 55,
	RED: 403,
	GREEN: 404,
	YELLOW: 405,
	BLUE: 406,
	BACKWARD: 412,
	FORWARD: 417,
	BACK: 461,
	PLAY_PAUSE: 463,
};

export default KEY_CODES;
