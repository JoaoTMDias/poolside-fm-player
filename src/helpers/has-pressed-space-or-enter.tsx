import { KEY_CODES } from "./key-codes";

/**
 * Checks if the user has pressed space or enter
 *
 * @param {number} keyCode
 * @returns {boolean}
 */
function hasPressedSpaceOrEnter(keyCode: number) {
	if (keyCode === KEY_CODES.ENTER || keyCode === KEY_CODES.SPACE) {
		return true;
	}

	return false;
}

export default hasPressedSpaceOrEnter;
