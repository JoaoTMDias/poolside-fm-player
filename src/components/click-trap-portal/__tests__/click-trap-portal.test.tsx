import React from "react";
import { mount } from "enzyme";
import { KEY_CODES, findByTestAttr } from "helpers";
import ClickTrapPortal from "..";

describe("<ClickTrapPortal />", () => {
	it("should render the portal", () => {
		const wrapper = mount(
			<div id="app-portal">
				<ClickTrapPortal onClickToClose={() => {}}>
					<p>Test</p>
				</ClickTrapPortal>
			</div>
		);

		expect(wrapper).toMatchSnapshot();
	});

	describe("should close the portal", () => {
		it("on click", () => {
			const onClickToCloseMock = jest.fn();

			const wrapper = mount(
				<div id="app-portal">
					<ClickTrapPortal onClickToClose={onClickToCloseMock}>
						<p>I am trapped inside the portal</p>
					</ClickTrapPortal>
				</div>
			);

			const button = findByTestAttr(wrapper, "component-portal-click-trap").first();
			button.simulate("click");

			expect(onClickToCloseMock).toHaveBeenCalled();
		});

		it("on press on ESC", () => {
			const map = {};

			window.addEventListener = jest.fn((event: any, callback: any) => {
				// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
				// @ts-ignore
				map[event] = callback;
			});

			const onClickToCloseMock = jest.fn();

			mount(
				<div id="app-portal">
					<ClickTrapPortal onClickToClose={onClickToCloseMock}>
						<p>I am trapped inside the portal</p>
					</ClickTrapPortal>
				</div>
			);

			// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
			// @ts-ignore
			map.keyup({ keyCode: KEY_CODES.ESC });

			expect(onClickToCloseMock).toHaveBeenCalled();
		});
	});

	it("should not close the portal on press on another key", () => {
		const map = {};

		window.addEventListener = jest.fn((event: any, callback: any) => {
			// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
			// @ts-ignore
			map[event] = callback;
		});

		const onClickToCloseMock = jest.fn();

		mount(
			<div id="app-portal">
				<ClickTrapPortal onClickToClose={onClickToCloseMock}>
					<p>I am trapped inside the portal</p>
				</ClickTrapPortal>
			</div>
		);

		// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
		// @ts-ignore
		map.keyup({ keyCode: KEY_CODES.ENTER });

		expect(onClickToCloseMock).not.toHaveBeenCalled();
	});

	it("should clear the global event listner on unmount", () => {
		window.removeEventListener = jest.fn();

		const onClickToCloseMock = jest.fn();

		const wrapper = mount(
			<div id="app-portal">
				<ClickTrapPortal onClickToClose={onClickToCloseMock}>
					<p>I am trapped inside the portal</p>
				</ClickTrapPortal>
			</div>
		);

		wrapper.unmount();

		expect(window.removeEventListener).toHaveBeenCalled();
	});
});
