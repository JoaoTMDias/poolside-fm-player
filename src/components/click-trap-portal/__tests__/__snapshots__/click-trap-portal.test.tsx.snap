// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClickTrapPortal /> should render the portal 1`] = `ReactWrapper {}`;
